acceptance_modelisation/bkg_collection.py

```diff
@@ -17,10 +17,23 @@ def __init__(self, bkg_dict: dict[float, BackgroundIRF] = None):
             bkg_dict : dict of gammapy.irf.BackgroundIRF
                 The collection of model in a dictionary with as key the zenith angle (in degree) associated to the model
         """
+        bkg_dict = bkg_dict or {}
         self.bkg_dict = {}
-        if not bkg_dict is None:
-            for k in bkg_dict.keys():
-                self[k] = bkg_dict[k]
+        for k, v in bkg_dict.items():
+            key = float(k)
+            self._check_entry(key, v)
+            self.bkg_dict[key] = v
+
+    @staticmethod
+    def _check_entry(key, v):
```

Comment:
The check on having all the model of the same type has been removed. If we maintain this, we need to handle this case at the interpolation step.

This also make me think it was also lacking a test to check that the axis are the same.

If these two conditions are not fulfilled the interpolation will failed for sure with likely a quite hard to understand.